netlogo-gui/src/main/app/PreferencesDialog.scala

```diff
@@ -0,0 +1,76 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.app
+
+import java.awt.{ BorderLayout, Frame }
+import java.awt.event.{ WindowAdapter, WindowEvent }
+import java.util.prefs.Preferences
+import javax.swing.{ BorderFactory, Box, BoxLayout, ImageIcon, JButton,
+  JComponent, JDialog, JLabel, WindowConstants, SwingConstants }
+
+import org.nlogo.awt.Positioning
+import org.nlogo.core.I18N
+import org.nlogo.swing.{ OptionDialog, RichAction, TextFieldBox, Utils => SwingUtils }
+import org.nlogo.swing.Implicits._
+
+object PreferencesDialog {
+  private val restartIcon =
+    new ImageIcon(classOf[PreferencesDialog].getResource("/images/forever2.gif"))
+}
+
+class PreferencesDialog(parent: Frame, preferences: Preference*)
```

Comment:
The actual preferences are injected from App. Is that OK, or should PreferencesDialog construct the list of preferences itself?